### PR TITLE
renode-unstable: init at 1.14.0+20231229gita76dac0ae, renode: init at 1.14.0

### PR DESCRIPTION
**Pull request by otavio**
### What is Renode?

Renode is a virtual development tool created by Antmicro for multi-node embedded networks (wired and wireless). Its purpose is to enable a scalable workflow for creating effective, tested, and secure IoT systems. Renode makes it fast, cost-effective, and reliable to develop, test, debug, and simulate unmodified software for IoT devices.

Supported architectures include:

- ARMv7 and ARMv8 Cortex-A, Cortex-R and Cortex-M
- x86
- RISC-V
- SPARC
- POWER
- Xtensa

The derivations are based on previous work done by @katyo and @thoughtpolice. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by otavio** [comment]
If @katyo and @thoughtpolice are interested in co-maintaining it, they are more than welcome. Would either of you be interested in doing so?

File: pkgs/by-name/re/renode/package.nix [hunk 1]
```diff
@@ -0,0 +1,109 @@
+{ stdenv
+, lib
+, fetchurl
+, autoPatchelfHook
+, writeScript
+
```
Comment (AndersonTorres) [inline pkgs/by-name/re/renode/package.nix:6]:
```suggestion
```

File: pkgs/by-name/re/renode/package.nix [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+, glibcLocales
+, python3Packages
+
```
Comment (AndersonTorres) [inline pkgs/by-name/re/renode/package.nix:9]:
```suggestion
```

File: pkgs/by-name/re/renode/package.nix [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+, gtk-sharp-2_0
+, gtk2-x11
+, screen
+
```
Comment (AndersonTorres) [inline pkgs/by-name/re/renode/package.nix:13]:
```suggestion
```

File: pkgs/by-name/re/renode/package.nix [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+, unstableVersion ? false
```
Comment (AndersonTorres) [inline pkgs/by-name/re/renode/package.nix:14]: A better name:

```suggestion
, buildUnstable ? false
```

Or even better, 

```suggestion
, buildVariant ? "stable" # can be "unstable" or "stable"
```

File: pkgs/by-name/re/renode/package.nix [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+stdenv.mkDerivation (finalAttrs: {
+  pname = "renode";
+  version = "1.14.0";
+
+  src = fetchurl {
+    url = "https://builds.renode.io/renode-${finalAttrs.version}.linux-portable.tar.gz";
+    hash = "sha256-1wfVHtCYc99ACz8m2XEg1R0nIDh9xP4ffV/vxeeEHxE=";
+  };
+
+  passthru.updateScript =
```
Comment (AndersonTorres) [inline pkgs/by-name/re/renode/package.nix:38]: Usually it goes at the end of expression before meta

File: pkgs/by-name/re/renode/package.nix [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+  propagatedBuildInputs = [
+    gtk2-x11
+    gtk-sharp-2_0
+    screen
+  ];
+
+  strictDeps = true;
+
+  installPhase = ''
```
Comment (AndersonTorres) [inline pkgs/by-name/re/renode/package.nix:72]: runHooks

File: pkgs/by-name/re/renode/package.nix [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    mkdir -p $out/{bin,libexec/renode}
+
+    mv * $out/libexec/renode
+    mv .renode-root $out/libexec/renode
+    chmod +x $out/libexec/renode/*.so
+
+    cat > $out/bin/renode <<EOF
```
Comment (AndersonTorres) [inline pkgs/by-name/re/renode/package.nix:79]: I believe makeWrappers are more apropriate.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/re/renode-unstable/package.nix [hunk 8]
```diff
@@ -0,0 +1,17 @@
+{ renode
+, fetchurl
+, unstableVersion ? true
+}:
+
+let
+  renode' = renode.override { inherit unstableVersion; };
+in
+renode'.overrideAttrs (finalAttrs: _: {
```
Comment (AndersonTorres) [inline pkgs/by-name/re/renode-unstable/package.nix:9]:
```suggestion
(renode.override { 
  inherit unstableVersion; 
}).overrideAttrs (finalAttrs: _: {
```

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**